test/parser.js

```diff
@@ -253,6 +253,20 @@ describe('Parser', function() {
     });
   });
 
+  it('should respect timeout option if the socket does not connect', function(done) {
+    var server = HTTP.createServer(function(req, res) {});
```

Comment:
I don't think we need to `createServer` here, since we're just connecting to a dead URL.

Reply:
For sure. Although I'm wondering if we need to spoof ECONNREFUSED in some way? I'm going to double check tonight and see if I can ensure it. Travis is failing and I'm wondering if it has to do with not correctly spoofing the error.

Reply:
Took this test case out since your suggestion to directly handle the timeout covers it as well, and I don't know if there's a way to reliably simulate ECONNREFUSED in the test suite.